### ISSUE TITLE
iTerm2

### ISSUE THREAD

**Issue by aristidesfl** [lra/mackup#988]
Even though iTerm2 can be configured to load settings from a specific location, it doesn't come enabled by default, and it doesn't do it seamlessly (like via iCloud)

What if every application had capabilities to sync their configuration? Would mackup be redundant? 

As an example, homebrew provides yet another way to install applications, even though there are at least 2 other ways.

Is there any technical reason which prevents iTerm2 from being supported, or is it the guideline?

**Comment by julienbourdeau** [comment]
And setting up each applications one by one is much more complicated than running one command. Mackup is a great tool, there should be as many _recipe_ as possible.

**Comment by vitorgalvao** [comment]
> What if every application had capabilities to sync their configuration? Would mackup be redundant?

Yes, pretty much. Especially since the method mackup uses is far from reliable and incapable of dealing with many apps (like every Mac App Store app).

> setting up each applications one by one is much more complicated than running one command. 

If the app uses iCloud Sync, you don’t even need to run a single command; it works without setup. When they don’t, it’s usually just a single `defaults write` command away, anyway, which is far more reliable.

**Comment by godbout** [comment]
@vitorgalvao any chance to detail a bit more what is that ```defaults write``` command? I mean, how can we make the settings of the MAS apps sync. I've forked mackup for a couple of apps and I've found out that in each case, the linked settings are removed, and a new files is created on top of them.

**Comment by vitorgalvao** [comment]
@godbout I meant `defaults read`.

Mackup doesn’t work for MAS apps precisely because it symlinks. You’d need to overwrite/copy the preference files and then do `defaults read {{app_bundle_id}}`.

**Comment by godbout** [comment]
@vitorgalvao ok, so some changes needed in how Mackup handles the things. Got it. Thanks!

**Comment by fharper** [comment]
@lra you can close this issue as iTerm2 is now supported.
